sources/ghostdog/src/main.rs

```diff
@@ -75,11 +93,43 @@ where
     Ok(device_type.to_string())
 }
 
+/// Finds the device name using the nvme-cli
+fn find_device_name(path: String) -> Result<String> {
+    // nvme-cli writes the binary data to STDOUT
+    let output = Command::new(NVME_CLI_PATH)
+        .args(["id-ctrl", &path, "-b"])
+        .output()
+        .context(error::NvmeCommandSnafu { path: path.clone() })?;
+
+    parse_device_name(&output.stdout, path)
+}
+
+/// Parses the device name from the binary data returned by nvme-cli
+fn parse_device_name(device_info: &[u8], path: String) -> Result<String> {
+    // Bail out if the data returned isn't complete
+    ensure!(
+        device_info.len() == NVME_IDENTIFY_DATA_SIZE,
+        error::InvalidDeviceInfoSnafu { path }
+    );
+
+    // The vendor data is stored at the last 1024 bytes
+    // The device name is stored at the first 32 bytes of the vendor data
+    let device_name =
+        &device_info[NVME_IDENTIFY_DATA_SIZE - 1024..NVME_IDENTIFY_DATA_SIZE - 1024 + 32];
+
+    Ok(String::from_utf8_lossy(device_name).trim_end().to_string())
+}
+
 /// Print the device type in the environment key format udev expects.
 fn emit_device_type(device_type: &str) {
     println!("BOTTLEROCKET_DEVICE_TYPE={}", device_type);
 }
 
+/// Print the device name in the environment key format udev expects.
+fn emit_device_name(device_name: &str) {
+    println!("XVD_DEVICE_NAME={}", device_name)
```

Comment:
Per https://docs.aws.amazon.com/ebs/latest/userguide/nvme-ebs-volumes.html I believe this is fine because:

* Device names are at most 32 bytes long, and unused trailing bytes are space characters.
* Device names do not contain any characters that would require quoting or escaping.

Therefore this will give us a valid environment key assignment of the form XVD_DEVICE_NAME=/some/stuff and udev will be happy.